.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,78 @@
+name: build and validate
+on:
+  push:
+    branches:
+      - main
+  pull_request_target:
+    branches: 
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
+      
+      - name: create checksum file
+        uses: hypertrace/actions/checksum@main
+
+      - name: Cache packages
+        id: cache-packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/checksum.txt') }}
+          restore-keys: |
+            gradle-packages-${{ runner.os }}-${{ github.job }}
+            gradle-packages-${{ runner.os }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_USER }}
+          password: ${{ secrets.DOCKERHUB_READ_TOKEN }}
+      
+      - name: Build with Gradle
```

Comment:
Before rolling out across all repos, could we address the two pieces of the existing CI config that didn't get ported:
- Reporting test results (already did this as part of the ui port - example here https://github.com/hypertrace/hypertrace-ui/commit/20072af17d37c0f92ea79dd9b212457e5e63d204#diff-bc668a2c9f2299cef15b222055b4b4d5311646caec2e7610e540cee18ae9b948R85-R90)
- A gradle action that standardizes the CLI flags to correctly set parallelism, logging etc. for CI

Reply:
First one, I can add easily. Little confused on second one here. I explored a path to create gradle composite action based on arguments but composite actions don't support if conditions as of now so I dropped that idea.

Reply:
I was basically just thinking something kind of like below (not sure if this is quite right, but gives the idea)
```yaml
name: 'Gradle Command'
inputs:
  args:
    required: true
runs:
  using: "composite"
  steps: 
    - run: |
          ./gradlew ${{ inputs.args }} --info --max-workers=2 -Dorg.gradle.jvmargs=-Xmx2g -Dorg.gradle.console=plain --continue
      shell: bash
```

Reply:
Okay. I can take a stab at this. seems straightforward. But if we are looking for gradle actions anyways can we also look at already available actions like: 
- https://github.com/marketplace/actions/gradle-command
- https://github.com/burrunan/gradle-cache-action

or we want to do something like above only?

Reply:
So the other two tasks solve different problems - mainly (from my understanding after a quick looko) resolution of gradle itself and caching. I think ideally, we'd incorporate them into our gradle action too, but those aren't pain points we have right now - the more pressing need is an ability to propagate the cli flags we want for gradle without having to hardcode them at each invocation.

Reply:
I created gradle action and tested it here: https://github.com/hypertrace/attribute-service/pull/65

Reply:
reporting test result action is not able to parse this reports: https://github.com/hypertrace/attribute-service/runs/1642325282?check_suite_focus=true#step:10:1

Reply:
@aaron-steinfeld , can you check once? I don't see that as critical at the moment, because codecov is reporting anyways.

Reply:
The test reports is meant to report test results (junit) but is being given paths to test coverage (jacoco). Try changing `./**/jacoco/test/*.xml` to `./**/build/test-results/**/*.xml`

Reply:
> I don't see that as critical issue at the moment, because codecov is reporting anyways.

It's not critical, but different tools/jobs. If a test fails, codecov will never run, and even if it did, only conveys covered lines of code rather than test results. This step surfaces the results of the tests so a failed test is directly surfaced on the PR rather than buried somewhere in the logs of the test action.

Reply:
Thanks, this is resolved.